apps/framework-cli/src/framework/python/scripts/blocks_runner.py

```diff
@@ -0,0 +1,137 @@
+import asyncio
+from dataclasses import dataclass
+from clickhouse_connect import get_client
+from importlib import import_module
+import argparse
+import os
+import sys
+
+
+parser = argparse.ArgumentParser(description='Run blocks')
+parser.add_argument('blocks_dir_path', type=str,
+                    help='Path to the blocks directory')
+parser.add_argument('clickhouse_db', type=str, help='Clickhouse database name')
+parser.add_argument('clickhouse_host', type=str, help='Clickhouse host')
+parser.add_argument('clickhouse_port', type=int, help='Clickhouse port')
+parser.add_argument('clickhouse_username', type=str,
+                    help='Clickhouse username')
+parser.add_argument('clickhouse_password', type=str,
+                    help='Clickhouse password')
+parser.add_argument('clickhouse_use_ssl', type=str, help='Clickhouse use SSL')
+
+
+args = parser.parse_args()
+
+interface = 'http' if args.clickhouse_use_ssl == "false" else 'https'
+host = args.clickhouse_host
+port = args.clickhouse_port
+db = args.clickhouse_db
+user = args.clickhouse_username
+password = args.clickhouse_password
+blocks_dir_path = args.blocks_dir_path
+
+sys.path.append(blocks_dir_path)
+
+# TODO: Handle retries
+class DependencyError(Exception):
+    pass
+
+
+def get_file_name(path):
+    return os.path.splitext(os.path.basename(path))[0]
+
+
+def get_blocks_from_file(path):
+    # Make sure file path is in sys.path
+    block = import_module(get_file_name(path), package=blocks_dir_path)
+
+    # Find the Blocks definition in the module
+    blocks_def = block.Blocks
+
+    # Find the Blocks objects in the module
+    blocks_objs = [obj for obj in dir(block) if isinstance(
+        getattr(block, obj), blocks_def)]
+    
+
+    # Make sure there is exactly one Blocks object in the modules
+    if len(blocks_objs) == 0:
+        raise ValueError(f"No Blocks object found in {path}")
+
+    if len(blocks_objs) > 1:
+        raise ValueError(f"Multiple Blocks objects found in {path}")
+
+    return getattr(block, blocks_objs[0])
+
+
+def walk_dir(dir, file_extension):
+    file_list = []
+
+    for root, dirs, files in os.walk(dir):
+        for file in files:
+            if file.endswith(file_extension):
+                file_list.append(os.path.join(root, file))
+
+    return file_list
```

Comment:
Feels like some of this logic could be shared with TS right?

Reply:
Yea I think so. Added a ticket for it: https://linear.app/514/issue/514-1107/investigate-one-blocks-process-per-file